samples/listFilesPaginate.js

```diff
@@ -0,0 +1,54 @@
+// Copyright 2020 Google LLC
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//      http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+function main(bucketName = 'my-bucket') {
+  // [START storage_list_files]
+  /**
+   * TODO(developer): Uncomment the following line before running the sample.
+   */
+  // const bucketName = 'Name of a bucket, e.g. my-bucket';
+
+  // Imports the Google Cloud client library
+  const {Storage} = require('@google-cloud/storage');
+
+  // Creates a client
+  const storage = new Storage();
+
+  async function listFilesPaginated() {
+    // Lists files in the bucket
+    let getFilesOptions = {
+      maxResults: 1,
+    };
+    let allFiles = [];
+    do {
+      const [files, nextQuery] = await storage
+        .bucket(bucketName)
+        .getFiles(getFilesOptions);
+      allFiles = [...allFiles, ...files];
+
+      // nextQuery will have all the custom properties originally set in
+      // getFilesOptions together with nextPageToken already set.
+      getFilesOptions = nextQuery;
+    } while (getFilesOptions);
+
+    console.log('Files:');
+    allFiles.forEach(file => {
+      console.log(file.name);
+    });
```

Comment:
This is always tricky to explain. I think if a user is opting-out of auto-pagination, we don't need to show them how to recreate it. Just explaining the basic set of tools should suffice, maybe like:

```js
const [files, queryForPage2] = await bucket.getFiles({autoPaginate: false});

// Get the next set of results
if (queryForPage2 !== null) {
  const [files, queryForPage3] = await bucket.getFiles(queryForPage2);
}
```

Reply:
I was trying to make a sample that will run a few cycles.

Available options

1. `autoPaginate: false` - will return 1 page of up to 1000 results, in a simple sample there won't be a nextQuery
2. `maxResults: <more than 1 page worth>` - will autoPaginate on default up to set value. Not really relevant for a simple sample
3. `maxResults: <more than 1 page worth>` and `autoPaginate: false` - will return only 1 page worth of results
4. `maxResults: <less then 1 page>` (autoPaginate value irrelevant) - will return up to set value

Option 4 seemed possible in a simple sample.

WDYT?

Reply:
Do you mean the motivation is to make the sample testable? I don't think that's as important as explaining the concept, since we have extensive unit and system test coverage in the respective places (paginator / storage libs).

Reply:
Done